pkg/middleware/timeout_test.go

```diff
@@ -0,0 +1,94 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+// mockServerGRPCStream mocks grpc server stream and only returns context
+// otherwise noop, and we don't care whether it is called or not.
+type mockServerGRPCStream struct {
+	ctx context.Context
+}
+
+func (m mockServerGRPCStream) SetHeader(metadata.MD) error {
+	return nil
+}
+func (m mockServerGRPCStream) SendHeader(metadata.MD) error {
+	return nil
+}
+func (m mockServerGRPCStream) SetTrailer(metadata.MD) {}
+func (m mockServerGRPCStream) Context() context.Context {
+	return m.ctx
+}
+func (m mockServerGRPCStream) SendMsg(any) error {
+	return nil
+}
+func (m mockServerGRPCStream) RecvMsg(any) error {
+	return nil
+}
+
+func TestNewUnaryTimeoutInterceptor(t *testing.T) {
+	timeoutInterceptor := TimeoutInterceptor{
+		timeout: 5 * time.Millisecond,
+		logger:  logger.NewNoopLogger(),
+	}
+
+	handler := func(ctx context.Context, req any) (any, error) {
+		goFuncInit := make(chan struct{})
```

Comment:
```suggestion
		select {
		case <-time.After(20 * time.Millisecond):
			return nil, nil
		case <-ctx.Done():
			return nil, ctx.Err()
		}
```
Here and below in the streaming test, all of the code in this grpc handler can be simplified to the simple snippet above.